src/include/migraphx/shape.hpp

```diff
@@ -248,6 +313,10 @@ struct shape
 
     const std::vector<shape>& sub_shapes() const;
 
+    /*!
+     * Returns size of the data buffer.
+     * Assuming a packed shape, returns maximum size of the data buffer for dynamic shape.
```

Comment:
It only assumes packed shape for dynamic shapes, which is not completely clear in this comment.

Reply:
@CharlieL7 any update to what @pfultz2 suggested? Also, I would change the wording to something like "number of elements" in the data buffer, since size may imply something like bytes which this is not doing.